api-server/src/main.rs

```diff
@@ -16,69 +20,63 @@ const DEFAULT_LOGGING_CONFIG: &str = "info,tower_http=trace";
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+enum RunMode {
+    LocalWorkspace,
+    RemoteS3Url,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
 #[tokio::main]
 async fn main() -> Result<(), hyper::Error> {
     init_logging();
 
-    let mut b = dill::CatalogBuilder::new();
-
-    b.add::<infra::QueryServiceImpl>();
-    b.bind::<dyn domain::QueryService, infra::QueryServiceImpl>();
-
-    b.add::<infra::LocalDatasetRepositoryImpl>();
-    b.bind::<dyn domain::LocalDatasetRepository, infra::LocalDatasetRepositoryImpl>();
-
-    b.add::<infra::RemoteRepositoryRegistryImpl>();
-    b.bind::<dyn domain::RemoteRepositoryRegistry, infra::RemoteRepositoryRegistryImpl>();
-
-    b.add::<infra::DatasetFactoryImpl>();
-    b.bind::<dyn domain::DatasetFactory, infra::DatasetFactoryImpl>();
-
-    b.add::<infra::SyncServiceImpl>();
-    b.bind::<dyn domain::SyncService, infra::SyncServiceImpl>();
-
-    b.add::<infra::SearchServiceImpl>();
-    b.bind::<dyn domain::SearchService, infra::SearchServiceImpl>();
-
-    // TODO: Externalize configuration
-    b.add_value(infra::IpfsGateway {
-        url: Url::parse("http://localhost:8080").unwrap(),
-        pre_resolve_dnslink: true,
-    });
-    b.add_value(kamu::infra::utils::ipfs_wrapper::IpfsClient::default());
-
     let matches = cli_parser::cli(BINARY_NAME, VERSION).get_matches();
 
     let repo_url = matches.get_one::<Url>("repo-url").cloned();
+    let local_repo_path = matches.get_one::<PathBuf>("local-repo").cloned();
+
+    let catalog = if let (Some(_), Some(_)) = (&repo_url, &local_repo_path) {
```

Comment:
Nitpick: A `match` block might be more readable that `if let / else if let` branching.